pkg/kubelet/cm/dra/plugin/plugin.go

```diff
@@ -169,42 +176,37 @@ func (h *RegistrationHandler) validateVersions(
 	if existingPlugin.highestSupportedVersion.LessThan(newPluginHighestVersion) {
 		return newPluginHighestVersion, nil
 	}
-	return nil, errors.New(
-		log(
-			"%s for DRA plugin %q failed. Another plugin with the same name is already registered with a higher supported version: %q",
-			callerName,
-			pluginName,
-			existingPlugin.highestSupportedVersion,
-		),
-	)
-}
-
-func deregisterPlugin(pluginName string) {
-	draPlugins.delete(pluginName)
+	return nil, fmt.Errorf("%s DRA plugin %q: another plugin with the same name is already registered with a higher supported version: %q < %q", what, pluginName, newPluginHighestVersion, existingPlugin.highestSupportedVersion)
 }
 
 // DeRegisterPlugin is called when a plugin has removed its socket,
 // signaling it is no longer available.
 func (h *RegistrationHandler) DeRegisterPlugin(pluginName string) {
-	klog.InfoS("DeRegister DRA plugin", "name", pluginName)
-	deregisterPlugin(pluginName)
-	h.controller.removePlugin(pluginName)
+	if p := draPlugins.delete(pluginName); p != nil {
+		logger := klog.FromContext(p.ctx)
+		logger.V(3).Info("Deregister DRA plugin", "endpoint", p.endpoint)
+
+		// Did exist before. Let's clean up ResourceSlices for it because
+		// it might have died without doing that itself and might not
+		// come back.
+		go h.wipeResourceSlices(pluginName)
+
+		return
+	}
+
+	logger := klog.FromContext(h.ctx)
+	logger.V(3).Info("Deregister DRA plugin not necessary, was already removed")
 }
 
 // ValidatePlugin is called by kubelet's plugin watcher upon detection
 // of a new registration socket opened by DRA plugin.
 func (h *RegistrationHandler) ValidatePlugin(pluginName string, endpoint string, versions []string) error {
 	klog.InfoS("Validate DRA plugin", "name", pluginName, "endpoint", endpoint, "versions", strings.Join(versions, ","))
 
-	_, err := h.validateVersions("ValidatePlugin", pluginName, versions)
+	_, err := h.validateVersions("validating", pluginName, versions)
 	if err != nil {
-		return fmt.Errorf("validation failed for DRA plugin %s at endpoint %s: %+v", pluginName, endpoint, err)
+		return err
```

Comment:
Why remove the wrapper?

Reply:
Because the error created by `validateVersions` was already long and included the plugin name. Now I am using simple errors there and wrap here.